File: tests/fixtures/edgecase_repo/python/pkg/module5.py
"""Module 5 of the generated Python layer."""

from __future__ import annotations

from pkg.module4 import Service4


class Base5:
    """Base class for service 5."""


class Service5(Base5):
    """Service 5."""

    def handle(self, payload):
        """Handle a payload."""
        self.validate(payload)
        return Service4().handle(payload)

    def validate(self, payload):
        """Validate a payload."""
        return payload is not None


def helper_5(value):
    """Module-level helper."""
    return value

File: tests/fixtures/edgecase_repo/c/module5.c
#include <stdio.h>
#include "module4.h"

struct Base5 {
    int name;
};

typedef struct Base5 Base5;

int handle_5(const char *payload) {
    validate_5(payload);
    printf("%s", payload);
    return 0;
}

int validate_5(const char *payload) {
    return payload != 0;
}
